packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,386 @@
+import '../base/file_system.dart';
+import '../base/process.dart';
+import '../base/utils.dart';
+import '../build_info.dart';
+import '../build_system/targets/ios.dart';
+import '../globals.dart';
+import '../macos/cocoapod_utils.dart';
+import '../macos/xcode.dart';
+import '../plugins.dart';
+import '../project.dart';
+import '../runner/flutter_command.dart' show DevelopmentArtifact, FlutterCommandResult;
+import 'build.dart';
+
+class BuildIOSFrameworkCommand extends BuildSubCommand {
+  BuildIOSFrameworkCommand({this.aotBuilder, this.bundleBuilder}) {
+    usesTargetOption();
+    usesFlavorOption();
+    usesPubOption();
+    argParser
+      ..addFlag('debug',
+        negatable: true,
+        defaultsTo: true,
+        help: 'Whether to produce a framework for the debug build configuration. '
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('profile',
+        negatable: true,
+        defaultsTo: true,
+        help: 'Whether to produce a framework for the profile build configuration. '
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('release',
+        negatable: true,
+        defaultsTo: true,
+        help: 'Whether to produce a framework for the release build configuration. '
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('universal',
+        help: 'Produce universal frameworks that include all valid architectures. '
+              'This is true by default.',
+        defaultsTo: true,
+        negatable: true
+      )
+      ..addFlag('xcframework',
+        help: 'Produce xcframeworks that include all valid architectures (Xcode 11 or later).',
+      )
+      ..addOption('output',
+        abbr: 'o',
+        valueHelp: 'path/to/directory/',
+        defaultsTo: fs.path.join(fs.currentDirectory.path, 'build', 'ios', 'framework'),
+        help: 'Location to write the frameworks.',
+      );
+  }
+
+  AotBuilder aotBuilder;
+  BundleBuilder bundleBuilder;
+
+  @override
+  final String name = 'ios-framework';
+
+  @override
+  final String description = 'Produces a .framework directory for a Flutter module '
+      'and its plugins for integration into existing, plain Xcode projects.\n'
+      'This can only be run on macOS hosts.';
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{
+    DevelopmentArtifact.universal,
+    DevelopmentArtifact.iOS,
+  };
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final FlutterProject flutterProject = FlutterProject.current();
+    if (!flutterProject.isModule) {
+      throwToolExit('Building frameworks for iOS is only supported from a module.');
+    }
+
+    if (getCurrentHostPlatform() != HostPlatform.darwin_x64) {
+      throwToolExit('Building frameworks for iOS is only supported on the Mac.');
+    }
+
+    if (argResults['xcframework'] && xcode.majorVersion < 11) {
```

Comment:
Place this validation in `validateCommand()`

Reply:
oh cool, thanks